File: swift_pico_hw/src/arm.cpp
#include <chrono>
#include <cstdlib>
#include <memory>

#include "rc_msgs/srv/command_bool.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class Swift_Pico : public rclcpp::Node
{
    public:
        Swift_Pico() : Node("pico_controller") //initializing ros node with name pico_controller
        {
           
            client = this->create_client<rc_msgs::srv::CommandBool>("/drone/cmd/arming");
            while (!client->wait_for_service(1s)) {
                RCLCPP_INFO(this->get_logger(), "Arming service not available, waiting again,,,,");
            }

            auto request = std::make_shared<rc_msgs::srv::CommandBool::Request>();
            request->value = true;

            auto result = client->async_send_request(request);
            /*this->get_node_base_interface() is a method provided by the rclcpp::Node class. 
            It retrieves a pointer to the Node Base Interface, which is one of the internal interfaces that rclcpp::Node implements.
            The Node Base Interface is a foundational interface that provides access to certain core functionalities of the node, 
            such as spinning, managing execution, and interacting with the ROS 2 executor.When you call rclcpp::spin_until_future_complete or similar functions, 
            they require this base interface to operate on the node in a generic way. 
            It's used in contexts where the function doesn't need the full Node object but only the basic functionality provided by this interface.*/
             if (rclcpp::spin_until_future_complete(this->get_node_base_interface(), result) == rclcpp::FutureReturnCode::SUCCESS)
            {
                RCLCPP_INFO(rclcpp::get_logger("rclcpp"), result.get()->data.c_str());
            }         
        }

    private:
        
        rclcpp::Client<rc_msgs::srv::CommandBool>::SharedPtr client;

};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Swift_Pico>());
    rclcpp::shutdown();
    return 0;
}

File: swift_pico_hw/src/arm.py
#!/usr/bin/env python3

from rc_msgs.srv import CommandBool
import rclpy
from rclpy.node import Node

class Swift_Pico(Node):
	
    def __init__(self):
        super().__init__('pico_controller')  # initializing ros node with name pico_controller
        self.cli = self.create_client(CommandBool, "/drone/cmd/arming")
        while not self.cli.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('Arming service not available, waiting again,,,,')
        self.req = CommandBool.Request()

        future = self.send_request() # ARMING THE DRONE
        rclpy.spin_until_future_complete(self, future)
        response = future.result()
        self.get_logger().info(response.data)

    def send_request(self):
        self.req.value = True
        return self.cli.call_async(self.req)
    

def main(args=None):
	rclpy.init(args=args)
	swift_pico = Swift_Pico()

	try:
		rclpy.spin(swift_pico)
	except KeyboardInterrupt:
		swift_pico.get_logger().info('KeyboardInterrupt, shutting down.\n')
	finally:
		swift_pico.destroy_node()
		rclpy.shutdown()

if __name__ == '__main__':
	main()